src/System.Security.Cryptography.Encoding/src/System/Security/Cryptography/AsnEncodedDataCollection.cs

```diff
@@ -85,7 +85,16 @@ void ICollection.CopyTo(Array array, int index)
 
         public void CopyTo(AsnEncodedData[] array, int index)
         {
-            ((ICollection)this).CopyTo(array, index);
+            // Need to do the argument validation ourselves as AsnEncodedDataCollection throws ArgumentOutOfRangeException where List<>.CopyTo() throws
+            // ArgumentException.
+            if (array == null)
+                throw new ArgumentNullException(nameof(array));
+            if (index < 0 || index >= array.Length)
+                throw new ArgumentOutOfRangeException(nameof(index), SR.ArgumentOutOfRange_Index);
+            if (Count > array.Length - index)
```

Comment:
This last check would be caught by the call to `_list.CopyTo()` with the same argument type and message if it failed, so it could be left out as an optimisation.